### ISSUE TITLE
[Bug]: Major performance issue while loading new pages

### ISSUE THREAD

**Issue by Carbocode** [Floorp-Projects/Floorp#1416]
### Pre-Submission Checklist

- [X] I have checked the existing/closed issues to ensure this issue has not already been reported.
- [x] I am using the latest supported version of the browser.
- [X] I have tried disabling all browser extensions/add-ons to rule out interference from these.
- [X] I have tried clearing the browser cache and cookies.
- [X] I have checked with a new profile and the issue still occurs.
- [X] I have provided detailed step-by-step instructions on how to reproduce the issue.
- [X] I have included relevant screenshots or console outputs.
- [X] I have checked that this problem is not occurring in the latest version of Firefox ESR.
- [X] This issue is specific to this browser and does not occur in other browsers.

### Bug Summary

When opening a new tab, the browser's CPU usage spikes to 170-180%, causing the entire browser to freeze and other open pages to crash.

### Reproduction Steps

1. Opening a web page
2. Waiting for the crash

### Expected Behavior

Loading the website without issues

### Actual Behavior

When opening a new tab or launching an extension (Bitwarden), the browser's CPU usage spikes to 170-180%, causing the entire browser to freeze and other open pages to crash, making the browser impossible to use.

Normally, my browser and computer handle multiple heavy tabs without any slowdowns or bugs. 

Few days ago, a similar issue occurred with Workspaces, which drastically slowed down page loading. However, the issue stopped once Workspaces were disabled.

### Screenshots

<img width="1685" alt="Screenshot 2024-09-05 at 7 55 11 PM" src="https://github.com/user-attachments/assets/5d764391-80d6-46a8-b837-05597a3ed735">

### Browser Version

11.18.0 (64-bit)

### Operating System

MacOS Ventura

### Additional Context

PC: Mac M2
Extensions: Bitwarden, uBlockOrigin, CanvasBlocker, SkipRedirect, SmartRefer, MultiAccount Container

**Comment by hikkamorii** [comment]
Experiencing same issue on OpenSUSE, both with Flatpak and OBS packages.

**Comment by CutterKnife** [comment]
What settings do you use for New Tab?

**Comment by Carbocode** [comment]
@CutterKnife can't say, because I didn't change anything related to tabs

**Comment by creeper-0910** [comment]
Try disabling the extension.

**Comment by Carbocode** [comment]
@creeper-0910 In case you missed:
<img width="940" alt="Screenshot 2024-09-06 at 9 51 01 AM" src="https://github.com/user-attachments/assets/e1b4f8ae-5bb1-45c7-a1e6-a69920ac070a">

**Comment by creeper-0910** [comment]
Sorry, I am checking just in case some of you have checked the box but have not actually checked it.

**Comment by CutterKnife** [comment]
> @CutterKnife can't say, because I didn't change anything related to tabs

I have not yet been able to reproduce this problem on my M1 Mac with macOS Sonoma...
It maybe due to differences in settings or installed add-ons.

**Comment by Carbocode** [comment]
By now the issue is resolved by disabling the "Web Apps" feature. I didn't have any installed, but the feature was enabled. 
I wasn’t sure at first, but I suspected the loading problem might have been related to SPAs. However, since nowadays almost every website is built using SPA frameworks, I couldn’t be certain.

I'm sure this is not the root of the problem. Because, although to a lesser extent, the issue was present with "Workspaces" feature enabled. It was resolved after disabling it, but as you can see, it reappeared a week later and was then fixed by disabling another feature.

Currently, the issue is resolved, but we'll have to see if it comes back in a few days, and I'll end up having to disable yet another feature.

At this rate, it might be quicker to just switch back to vanilla Firefox 🤣. \s

**Comment by celozzip** [comment]
yeah i'm gonna try out zen. i dunno when the last update came out, but i just updated floorp today, and the web apps button is broken, the bookmarks toolbar suddenly doesnt work...

**Comment by uniqueUsernameIsNotAvailable** [comment]
Can confirm. Latest major Floorp update leads to CPU load peak when opening tabs, or switching to another.
Browser starts to lag a lot, easiest solution - reopen browser and you will get a approx. 2 hours of stable work.

Disabled Web Apps, will continue investigation.

**Comment by surapunoyousei** [comment]
Web apps are not implemented to affect new tabs in the first place.

PWAs only open windows with a single tab, so we need to track down the cause

**Comment by surapunoyousei** [comment]
> Can confirm. Latest major Floorp update leads to CPU load peak when opening tabs, or switching to another. Browser starts to lag a lot, easiest solution - reopen browser and you will get a approx. 2 hours of stable work.
> 
> Disabled Web Apps, will continue investigation.

CPU load increases with each update, which is natural since web functionality is increasing.

**Comment by Carbocode** [comment]
> Web apps are not implemented to affect new tabs in the first place.
> 
> PWAs only open windows with a single tab, so we need to track down the cause

Well, but the browser offers to install it. So having the feature enabled surely does something in background. I'm sure the related code isn't the one behind those performance issues, but it can lead to the real bug. Considering also the workspaces feature did the same thing in my case.

**Comment by surapunoyousei** [comment]
I have never used similar code for workspaces and PWAs so I don't see the relevance.

I have macOS and have not experienced this.

**Comment by Carbocode** [comment]
> I have never used similar code for workspaces and PWAs so I don't see the relevance.
> 
> I have macOS and have not experienced this.

ah ok, I'm just a skizo

**Comment by surapunoyousei** [comment]
I am not fluent in English, so I would appreciate a little more information.

I would like to fix the bugs in Floorp as much as possible.

**Comment by uniqueUsernameIsNotAvailable** [comment]
@surapunoyousei 
How to reproduce major performance issue with new update: 
work in browser for several hours - try to close any tab or open new - get a 5 seconds freeze.

I did a profiling, you can see 99% of cpu core usage because of memcpy(), which is strange.

1 file - browser was in work for some time, tried to close random tab - got "not responding"
2 file - freshly opened browser - no lags.

[lags.json.gz](https://github.com/user-attachments/files/16919078/lags.json.gz)
[freshBrowser.json.gz](https://github.com/user-attachments/files/16919080/freshBrowser.json.gz)

**Comment by uniqueUsernameIsNotAvailable** [comment]
Don't want to rollback, because latest version fixed "Show Sidebar over website content" feature for me.

Also, I have disabled Web apps and Workspaces

**Comment by vKrishnat** [comment]
I Can confirm similar observations on WIndows 11 23H2 (22631.4112).  

The browser is stable and responsive for an hour or 2..
Closing any tab or window will make other Tabs or browser window respond slowly/Freeze.
CPU usage goes above 140% 
Afterwards becomes sluggish and at times "not responding"
Either restart or kill the process tree. This gives another 2 hours to work. 

Disabling extensions and/or resetting settings to default doesn't help.
Workspaces and containers are not used.

Issue started after version 11.17 update onwards.
Uninstalled and re-installed the Floorp, still the problem exists. 

Extensions : uBlock origin, Ghostery, Gesturfy.

**Comment by vKrishnat** [comment]
Here is the screenshot
![image](https://github.com/user-attachments/assets/a6c2ed3f-4f2b-42ea-82db-06f932c954b8)

**Comment by creeper-0910** [comment]
Do you have anti-virus software installed?

**Comment by vKrishnat** [comment]
> Do you have anti-virus software installed?

Yes. But the same antivirus was there prior to 11.17 update also.

**Comment by Carbocode** [comment]
So? Are we going to keep shifting the responsibility for the problem onto other things? Our computers have remained exactly the same, and if this issue appeared after a browser update, it can only be related to Floorp, not extensions, other software, or the hardware of our computers.

Now, I find myself heavily dependent on split view and the sidebar, yet the days keep passing, and all that’s happening here is trying to blame the problem on other things? At this point, I’d rather make compromises but at least have an acceptable user experience by switching to something else.

**Comment by creeper-0910** [comment]
As long as it is not reproduced in all environments, the influence of external software cannot be removed from consideration.

**Comment by surapunoyousei** [comment]
We are very sorry.

Floorp is basically written by me alone, and I am developing it in response to various requests.

Floorp was the first browser to introduce split views and workspaces based on Firefox, so there is not a lot of code to draw from like Zen or Firedoragon.

All the revenue generated by Floorp goes to development, and I continue to write code for free.

I don't have much time so I have to suspect other problems first.

**Comment by Carbocode** [comment]
Look, I’m not trying to undervalue your work at all. I really appreciate what you’re doing, and I think Floorp is a very important fork, that has brought new life to Firefox.

Unfortunately, even though this project has some influence, there aren’t many real users, and even fewer of us are active on GitHub and able to report issues.

We all have different devices with different operating systems and software, so I think it’s unlikely the issue is coming from outside. Ironically, the only things we might all have in common are some Electron apps running in the background.

**Comment by gunslingerfry** [comment]
Also seeing issues here on Arch Linux as of the latest updates. I see CPU spikes randomly but especially when opening a new tab and viewing a page.  Using htop I see the CPU spike to 150+ and the application halts. None of the subprocesses seem to be consuming the CPU so it has to be the main process doing something. I see no correlation to Electron apps. My floorp process pegs a core + seemingly at random. It's at 110% right now and text input is delayed.
![20240911_14h05m55s_grim](https://github.com/user-attachments/assets/17224663-95ad-4f3e-aac5-78f959b17c30)

**Comment by gunslingerfry** [comment]
@surapunoyousei very much appreciate the work you are doing. It is allowing me to avoid using edge but still get all that sweet vertical real estate on my monitor and without the buggy CSS hacks firefox requires. If there is anything I can provide to help let me know.

**Comment by gunslingerfry** [comment]
Interestingly it seems to be working better once I unloaded the one side tab I had loaded. Could it have something to do with how the side tabs are backgrounded?

**Comment by Carbocode** [comment]
<img width="434" alt="Screenshot 2024-09-16 at 9 49 38 AM" src="https://github.com/user-attachments/assets/faff190a-a01f-4d4f-99b3-28ecf7052330">
Just restarted the Browser as always to get normal performances, when this happened

**Comment by surapunoyousei** [comment]
If possible, it would be great to have a video up to the discharge that is malfunctioning!

**Comment by uniqueUsernameIsNotAvailable** [comment]
@surapunoyousei have you seen my reply with logs of low performance? [here](https://github.com/Floorp-Projects/Floorp/issues/1416#issuecomment-2335788353)

**Comment by surapunoyousei** [comment]
I read it, but nothing seemed useful on Floorp, where we only do front-end mods.

**Comment by gunslingerfry** [comment]
I checked the firefox bugzilla but I'm not finding anything related to new pages causing performance issues 😢  Maybe it's just a little timing thing that Floorp hits but Firefox doesn't. I've uninstalled for now because it's running about 10x slower than edge in Linux and it's a bit maddening. I'll reinstall fresh in a month or so and see how things go.

**Comment by 3DPrintSpace** [comment]
> If possible, it would be great to have a video up to the discharge that is malfunctioning!

I left Floorp open on my desktop overnight.  It was working fine.  This is what I came back to this morning.  Memory usage spikes and continues to grow.  Just opening a new tab, closing a tab, or opening a new page causes this behavior.

If you wait, the page will recover and load eventually.  However, interacting with that page will cause the cycle to occur again.  Closing Floorp and reopening fixes the issue.

Hope this video helps

https://github.com/user-attachments/assets/1a50dd96-3dc0-4bdf-bd94-627b5f50f194

**Comment by techfixes1** [comment]
On Windows here. Floorp has been really bogging down something shocking the last couple of weeks. When I switch to another tab it hangs for sometimes a couple of minutes or more. I turned off side tabs, hardware acceleration, web apps, turning off all extensions, still no change. The only thing that does work (temporarily) is dumping all the tabs and closing / reopening the browser. Strangely enough I'm not getting this on my PC at work, which has less memory than this one (24 GB)

**Comment by uniqueUsernameIsNotAvailable** [comment]
Uh, 11.17 version is soooo robust and performant. Guys, consider rollback and wait till investigation will finish. 
All the same, 11.18 is not so major for ordinary user

**Comment by guigirl42** [comment]
I am getting the same problems in my Floorp `11.18.1` in `ArchLinux KDEPlasma`:
- https://github.com/Floorp-Projects/Floorp/issues/1416
- https://github.com/Floorp-Projects/Floorp/issues/1440
- https://github.com/Floorp-Projects/Floorp/issues/1421

Initially Floorp behaves nicely fast, responsive, & low cpu usage & temperature. However after an hour or two of ongoing browsing, "something" goes badly wrong, such that lots of normal functions become **very very laggy**, & the **cpu % & temperature shoots up**. This is _not_ site-specific for me; i'm just using all my normal daily dozens of news & tech sites as usual. The only thing i can do is close Floorp & relaunch it, _but_... once these problems have begun, Floorp usually refuses to close immediately, instead lingering for a couple of minutes with fans spinning fast from high temperature, & high cpu%.

I'm sad to say that until this problem is solved, using Floorp is no longer nice, so i've gone back to FF Nightly to wait for a solution.

This only began happening for me with Floorp's change from FF ESR 115 ["good"] to FF ESR 128 ["bad"]. I've tried two profiles & both suffer the problem, ie, afaict it's a _browser_ problem, not a single bad _profile_ of mine.

**Comment by techfixes1** [comment]
This is giving me daily hell. I got so sick of Browser hopping and thought I'd found a permanent home with Floorp. Now it's been lagging badly **every single day** for weeks.

What version do I roll back to? (I'm on Windows 11) I'm ready to try anything, but don't want to give up on Floorp. I really hoped this buggy version could have been pulled before now.

**Comment by uniqueUsernameIsNotAvailable** [comment]
@techfixes1 try 11.16.0 it is pretty stable

**Comment by guigirl42** [comment]
Unfortunately [this bad bug](https://github.com/Floorp-Projects/Floorp/issues/1416#issuecomment-2379519517) _remains active_ in the newly released `v11.19.0`. Today i only had ~3 hours of decent browsing before once again everything became terribly sluggish & laggy, with high cpu% & temperature.

**Comment by Carbocode** [comment]
@guigirl42 still there, because they refuse to recognize it as a bug of their own, despite it happening in devices with different architecture, different operating system and only on Floorp.
Given that the problem is only being ignored, I will just switch to another browser

**Comment by guigirl42** [comment]
> I will just switch to another browser

Yep, i'm posting this from the lovely `Zen Browser`.  I would love though for this bug to be positively identified & solved, coz i still want Floorp to succeed.

**Comment by creeper-0910** [comment]
What happens if I disable ads on a new tab?

**Comment by techfixes1** [comment]
Still got the problem on the new v11.19.0 (on Windows 11)

Every hour or so I have to run this command from Administrator CMD prompt to get my browser usable again:    taskkill /IM floorp.exe /f

**Comment by 3DPrintSpace** [comment]
> If I disable the ads on the new tab, does that fix the problem?

I had tried this, but didn't seem to help.

**Comment by absol-0359** [comment]
In my case, disabling “Color the Tab Bar using the current website's favicon color” seems to have solved (or alleviated) the performance problem.
On Version 11.19.0, Windows 11 23H2 (22631.4317)

**Comment by creeper-0910** [comment]
Thanks for the information!
I will enable that feature and test it.

**Comment by creeper-0910** [comment]
> In my case, disabling “Color the Tab Bar using the current website's favicon color” seems to have solved (or alleviated) the performance problem. On Version 11.19.0, Windows 11 23H2 (22631.4317)

Just to confirm, did this fix the problem with opening new tabs as well?

**Comment by absol-0359** [comment]
>Just to confirm, did this fix the problem with opening new tabs as well?

Yes too.

**Comment by creeper-0910** [comment]
> > Just to confirm, did this fix the problem with opening new tabs as well?
> 
> Yes too.

Okay, thank you very much!

**Comment by RetroitKnife** [comment]
I also have the same problem and turning off "Color the Tab Bar using the current website's favicon color" fixed it for me as well.

I am using:
- Floorp v11.19.0 & v11.19.1
- Various addons, such as uBlock, Decentraleyes and Firefox Container

**Comment by Knewest** [comment]
> In my case, disabling “Color the Tab Bar using the current website's favicon color” seems to have solved (or alleviated) the performance problem. On Version 11.19.0, Windows 11 23H2 (22631.4317)

This fixed the issue for you. Bless your soul.

**Comment by melroy89** [comment]
Some websites (like Nextcloud might be a good example) are loading very badly within Floorp, the DOM processing is slow, mainly  "executing scripts" step (aka JS).

![image](https://github.com/user-attachments/assets/b6d274cb-c640-4f9b-b37a-f8ffbb8789af)

Here another Nextcloud server (other domain, with other plugins):

![image](https://github.com/user-attachments/assets/0385a81f-d2ea-459f-94c9-193907071e21)

In both cases the amount of plugins enabled are limited... 

For comparison the same site in Chromium:

![image](https://github.com/user-attachments/assets/2c1ef0c0-61ba-4038-ab64-02de2f88f499)

(Hot cache in both cases)

Hopefully Firefox & Floorp can improve DOM parsing & JS script execution.


And this was a good weather scenario, I also sometimes see Floorp struggling alot, and waiting for 20 or 30 seconds to get the page loaded (again due to JS).

**Comment by lmBored** [comment]
Have there been any solutions for this yet? Or this is expected to be fixed in floorp 12?

**Comment by Technetium1** [comment]
I would personally expect a fix in v12, especially if it's related to DOM parsing and script execution shown in the comment above yours. I was previously able to reproduce the problem with colored tabs causing performance issues on Linux and Windows (https://github.com/Floorp-Projects/Floorp/issues/1416#issuecomment-2408361956), but melroy89 seems to be onto something different.

**Comment by Poobslag** [comment]
I still encounter the same issue in Floorp 11.30.0, even with all plugins disabled. I regularly open 10 websites in new tabs to keep up with comments (Twitter, Bluesky, Patreon, etc...) and this used to happen instantaneously. Now it freezes the browser for about 5 seconds while Windows reports very high CPU/power usage.
